pkg/apis/core/validation/validation.go

```diff
@@ -5170,6 +5170,11 @@ func ValidatePodUpdate(newPod, oldPod *core.Pod, opts PodValidationOptions) fiel
 		allErrs = append(allErrs, field.Invalid(fldPath, newPod.Status.QOSClass, "Pod QoS is immutable"))
 	}
 
+	isPodResizeRequestValid := isPodResizeRequestValid(*oldPod)
```

Comment:
this might need to move to `PodValidationOptions`. TBH, I'm not sure what decides what goes in the validation options - maybe ask an API reviewer.

Reply:
I was trying to move this `isPodResizeRequestValid` function to PodValidationOptions and make it aligned with everything else there, but I soon realized that the opts that `ValidatePodUpdate` gets comes from the `GetValidationOptionsFromPodSpecAndMeta` function. We need to check if Resources are present in the Status however. Can PodValidationOptions have logic based on the PodStatus?

https://github.com/kubernetes/kubernetes/blob/8c7160205db67b250dece7cad5c6367debad2642/pkg/registry/core/pod/strategy.go#L149-L151